.github/workflows/end-to-end-tests.yml

```diff
@@ -0,0 +1,36 @@
+name: "end-to-end_tests"
```

Comment:
I suspect we'll end up piling all test jobs into one file, as we convert more runs to GHA, similar to https://github.com/matrix-org/synapse/blob/develop/.github/workflows/tests.yml. Perhaps let's do something similar and name this file just `tests.yml` or something similar generic, and then use a similar generic name here at the root level...?